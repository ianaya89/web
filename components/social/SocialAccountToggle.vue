<template>
  <div class="flex items-center">
    <switcher :value="isActivated" :disabled="isActivated" label="" @input="activate" />
    <avatar
      :src="this.$get(account, 'profilePicture.url', undefined)"
      :rounded="false"
      :icon-sign="$helpers.socialMediaIcon(account.type)"
      classes="h-12 w-12 mr-4"
    />
    <span class="font-serif text-grey">
      {{ account.name || account.url }}
    </span>
  </div>
</template>

<script>
import Avatar from '@/components/Avatar'
import Switcher from '@/components/Switcher'

export default {
  name: 'SocialAccountToggle',

  components: {
    Avatar,
    Switcher
  },

  props: {
    account: {
      type: Object,
      required: true
    }
  },

  computed: {
    isActivated() {
      return (
        ['youtube', 'instagram'].includes(this.account.type) ||
        this.account.isAuthorized
      )
    }
  },

  methods: {
    activate(val) {
      if (val) {
        window.location.replace(
          `${process.env.VUE_APP_BACKEND_URL}/auth/${
            this.account.type
          }?unit_id=${this.$route.params.id}&account_id=${this.account.id}`
        )
      }
    }
  }
}
</script>
