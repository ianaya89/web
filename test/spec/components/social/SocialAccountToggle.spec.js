import { shallowMount } from '@vue/test-utils'
import SocialAccountToggle from '@/components/social/SocialAccountToggle'

describe('SocialAccountToggle', () => {
  it('renders SocialAccountToggle', () => {
    const wrapper = shallowMount(SocialAccountToggle, {
      mocks: {
        $helpers: {
          socialMediaIcon: jest.fn()
        }
      },
      propsData: {
        account: {
          id: 1,
          name: 'Totally awesome page',
          url: 'https://facebook.com/awesome-page',
          profilePicture: {
            url: ''
          },
          type: 'facebook',
          isAuthorized: true
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('checks if account is activated', () => {
    const wrapper = shallowMount(SocialAccountToggle, {
      mocks: {
        $helpers: {
          socialMediaIcon: jest.fn()
        }
      },
      propsData: {
        account: {
          id: 1,
          name: 'Totally awesome page',
          url: 'https://facebook.com/awesome-page',
          profilePicture: {
            url: ''
          },
          type: 'facebook',
          isAuthorized: false
        }
      }
    })

    expect(wrapper.vm.isActivated).toBeFalsy()

    wrapper.setProps({
      account: {
        id: 1,
        name: 'Totally awesome page',
        url: 'https://facebook.com/awesome-page',
        profilePicture: {
          url: ''
        },
        type: 'facebook',
        isAuthorized: true
      }
    })

    expect(wrapper.vm.isActivated).toBeTruthy()
  })
})
