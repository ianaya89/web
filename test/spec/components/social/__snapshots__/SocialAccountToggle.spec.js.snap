// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAccountToggle renders SocialAccountToggle 1`] = `
<div
  class="flex items-center"
>
  <switcher-stub
    color="red"
    disabled="true"
    label=""
    rules=""
    value="true"
  />
   
  <avatar-stub
    avatarclasses=""
    classes="h-12 w-12 mr-4"
    src=""
    tooltipcolor="light"
    tooltipposition="top"
  />
   
  <span
    class="font-serif text-grey"
  >
    
    Totally awesome page
  
  </span>
</div>
`;
